torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -561,7 +561,13 @@ void TensorPipeAgent::pipeRead(
       return;
     }
 
-    std::vector<c10::Stream> streams = getStreamsFromPoolForDevices(devices_);
+    std::vector<c10::Stream> streams;
+    if (isStaticGroup_) {
+      streams = getStreamsFromPoolForDevices(devices_);
+    } else {
+      std::unique_lock<std::mutex> lock(groupMembershipMutex_);
```

Comment:
nit: since we don't need to lock and unlock here, a `lock_guard` might be better?

Reply:
Switched it with custom lock guard as suggested